### ISSUE TITLE
UnicodeDecodeError with CloudComputeCannonEngine

### ISSUE THREAD

**Issue by avirshup** [Autodesk/pyccc#3]
This is with pyccc 0.6.4.

To reproduce:

``` python
import pyccc
engine = pyccc.CloudComputeCannon('http://cloudcomputecannon.bionano.autodesk.com:9000')
def add_one_to_it(x): return x+1
cmd = pyccc.PythonCall(add_one_to_it, 5)
job = engine.launch('python:2.7-slim', cmd)
job.wait()
job.result
```

Raises:

```
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-6-5900baa395b4> in <module>()
      3 def add_one_to_it(x): return x+1
      4 cmd = pyccc.PythonCall(add_one_to_it, 5)
----> 5 job = mdt.compute.default_engine.launch('python:2.7-slim', cmd)
      6 job.wait()
      7 job.result

/usr/local/lib/python2.7/site-packages/pyccc/engines/base.pyc in launch(self, image, command, **kwargs)
     55         else:
     56             JobClass = Job
---> 57         return JobClass(self, image, command, **kwargs)  # creates and submits the job
     58 
     59     def submit(self, job):

/usr/local/lib/python2.7/site-packages/pyccc/python.pyc in __init__(self, engine, image, command, sendsource, **kwargs)
     73 
     74         super(PythonJob, self).__init__(engine, image, self.SHELL_COMMAND,
---> 75                                         **kwargs)
     76 
     77     def _get_python_files(self):

/usr/local/lib/python2.7/site-packages/pyccc/job.pyc in __init__(self, engine, image, command, name, submit, inputs, outputs, requirements, numcpus, runtime, on_status_update, when_finished)
    102         self.jobid = None
    103 
--> 104         if submit and self.engine and self.image: self.submit()
    105 
    106     kill = EngineFunction('kill')

/usr/local/lib/python2.7/site-packages/pyccc/job.pyc in submit(self, block)
    127 
    128     def submit(self, block=False):
--> 129         self.engine.submit(self)
    130         if block: self.wait()
    131 

/usr/local/lib/python2.7/site-packages/pyccc/engines/ccc.pyc in submit(self, job)
    106                                          cpus=job.numcpus,  # how is this the "minimum"?
    107                                          maxDuration=1000*job.runtime,
--> 108                                          workingDir='/workingdir')
    109 
    110         job.jobid = returnval['jobId']

/usr/local/lib/python2.7/site-packages/pyccc/jsonrpc.pyc in jsonRpcFunction(**kwargs)
     97                 response = requests.post(host,
     98                                          headers=headers,
---> 99                                          data=json.dumps(jsonRpcRequest))
    100 
    101             try:

/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.pyc in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, encoding, default, sort_keys, **kw)
    242         cls is None and indent is None and separators is None and
    243         encoding == 'utf-8' and default is None and not sort_keys and not kw):
--> 244         return _default_encoder.encode(obj)
    245     if cls is None:
    246         cls = JSONEncoder

/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.pyc in encode(self, o)
    205         # exceptions aren't as detailed.  The list call should be roughly
    206         # equivalent to the PySequence_Fast that ''.join() would do.
--> 207         chunks = self.iterencode(o, _one_shot=True)
    208         if not isinstance(chunks, (list, tuple)):
    209             chunks = list(chunks)

/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.pyc in iterencode(self, o, _one_shot)
    268                 self.key_separator, self.item_separator, self.sort_keys,
    269                 self.skipkeys, _one_shot)
--> 270         return _iterencode(o, 0)
    271 
    272 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

UnicodeDecodeError: 'utf8' codec can't decode byte 0x80 in position 0: invalid start byte
```

**Comment by avirshup** [comment]
This is a problem with the pickle file, where "some freedom is taken in the encodings of certain objects" (https://docs.python.org/2.7/library/pickle.html).

So this should just be treated as binary.

**Comment by avirshup** [comment]
Fixed in Autodesk/py-cloud-compute-cannon@adb46c24c004adcf6d911af3b8525990facdf444
